packages/store/src/patterns/patternMatchers.js

```diff
@@ -929,18 +1183,32 @@ const makePatternKit = () => {
 
   /** @type {MatchHelper} */
   const matchMapOfHelper = Far('match:mapOf helper', {
-    checkMatches: (specimen, [keyPatt, valuePatt], check) =>
-      checkKind(specimen, 'copyMap', check) &&
-      specimen.payload.keys.every((k, i) =>
-        checkMatches(k, keyPatt, check, `keys[${i}]`),
-      ) &&
-      specimen.payload.values.every((v, i) =>
-        checkMatches(v, valuePatt, check, `values[${i}]`),
-      ),
+    checkMatches: (
+      specimen,
+      [keyPatt, valuePatt, limits = undefined],
+      check,
+    ) => {
+      const { numMapEntriesLimit } = limit(limits);
+      return (
+        checkKind(specimen, 'copyMap', check) &&
+        check(
+          specimen.payload.keys.length <= numMapEntriesLimit,
```

Comment:
The format for all the specimen types should be described somewhere, and reference from the top of the matchers section

Reply:
Instead I wrote https://github.com/Agoric/agoric-sdk/issues/6148 as a reminder that we need this.